internal/api/utils.go

```diff
@@ -223,22 +223,17 @@ func TransformComponentUnits(jsonData datatypes.JSON) map[string]interface{} {
 		return nil
 	}
 
-	durationBased, ok := data["duration_based"].(map[string]interface{})
-	if !ok {
-		fmt.Printf("duration_based not found in JSON")
-	}
-
 	convertMemory := func(memory map[string]interface{}) error {
```

Comment:
Maybe not in the scope of this pull request, however if I understand the pattern then we are saving/naming a func block to `convertMemory` and later in the code calling that with `memory` as param. Considering the the size of the `TransformComponentUnits` I at least see we can at least move `convertMemory`, `hasMoreThanThreeDecimals`, `truncateToThreeDecimalPlaces` and `convertCPU` to separate functions. Even after moving these to separate func I still see the scope to divide this long func in smaller one.

I have one more thought, we have utils.go, and I feel we also have scope for separate package for TransformComponentUnits and related funcs.

Further to this, I understand the convert* funcs are very basic, it seems maybe not requiring the tests, however for coverage its good to have?

Reply:
The reason for housing all these smaller functions under single function is because of the are only used in the context `TransformComponentUnits`, I don't see these smaller functions getting re-used anywhere else. 
But I agree moving those functions outside will help with readability.
IMO a separate package may not be worth it, when it comes down to maintenance, cause of the limited scope.
Test are jolly good any day, but it will take some time for me to include them as a practice.